### PR TITLE
add double quotes to data selector values

### PR DESCRIPTION
**Pull request by Manuel-Suarez-Abascal**
**What was changed with this Pull Request?**

- Added `method` to check for `data attributes selectors` without `double quotes` & wrap `double quotes` around them.
- Updated `unit` tests.
- Fixed small typo 😉 

**Why was this change needed?**

It was to resolve a bug in the `Cypress Playground Selector` [here](https://github.com/cypress-io/cypress/issues/1884).

**Passing unit tests**

Screen Shot 2021-10-22 at 11 46 48 PM

### CODE REVIEW

File: src/getData.js [hunk 1]
```diff
@@ -31,13 +50,13 @@ export const getData = ( selectorType, attributes ) =>
...
-          return `[${nodeName}="${value}"]`;
+          return _setQuotesToAttrSelector(`[${nodeName}="${value}"]`);
         }
 
-        return `[${nodeName}=${value}]`;
+        return _setQuotesToAttrSelector(`[${nodeName}=${value}]`);
       }
 
-      return `[${nodeName}]`;
+      return _setQuotesToAttrSelector(`[${nodeName}]`);
```
Comment (chrisbreiding) [inline src/getData.js]: Maybe I'm missing something, but it seems this can be simplified a good bit. Should be able to get rid of the `if (needsQuote( value ) )` check and always add quotes around the value. Then `_setQuotesToAttrSelector` won't be necessary at all. Also don't have to worry about the case where there is no value.

**Review by chrisbreiding — CHANGES_REQUESTED** [review]

File: src/getData.js (hunk 1 above)
Comment (Manuel-Suarez-Abascal) [reply src/getData.js]: @chrisbreiding I have addressed your PR comments. Can you take another look it at?

> Also don't have to worry about the case where there is no value.

I removed one of the unit tests which it was covering this edge case.

Comment (BlueWinds) [reply src/getData.js]: @chrisbreiding - Can you explain to me why do don't need to worry about not having a value anymore? I'm a bit lacking on context for this, but it seems like it's still an important case to cover.

```
if ( nodeName === selectorType ) {
  return value ? `[${nodeName}="${value}"]` : `[${nodeName}]`
}
```

Comment (chrisbreiding) [reply src/getData.js]: Sorry, my feedback wasn't clear. I meant that the attribute-only, value-less case doesn't need special handling in regards to quotes, so it should remain unchanged in this PR.

**Review by chrisbreiding — CHANGES_REQUESTED** [review]
Please add back support for selectors that are only an attribute with no value.

**Comment by chrisbreiding** [comment]
Hey @Manuel-Suarez-Abascal, do you think you'll have a chance to address the review for this PR?

**Comment by Manuel-Suarez-Abascal** [comment]
> Hey @Manuel-Suarez-Abascal, do you think you'll have a chance to address the review for this PR?

Hi, @chrisbreiding I might take a look at it again during the weekend! 🚀  I am a bit busy with work this week 😬

File: src/getData.js (hunk 1 above)
Comment (Manuel-Suarez-Abascal) [reply src/getData.js]: @chrisbreiding I have addressed your PR comment. Could you take another look at it, please? Hopefully this time I got it right 😬 Also, all unit tests are passing.

Screen Shot 2021-11-15 at 2 27 25 AM

Comment (Manuel-Suarez-Abascal) [reply src/getData.js]: Also, maybe for future improvement, we can add some linting rules to match the `cypress` repo? I think this repo will benefit from some linting rules :)
